File: Passbook/MovieNight/ObjC/DTAVFoundationFunctions.m
#import "DTAVFoundationFunctions.h"

// helper function to convert interface orienation to correct video capture orientation
AVCaptureVideoOrientation
DTAVCaptureVideoOrientationForUIInterfaceOrientation(
                            UIInterfaceOrientation interfaceOrientation)
{
	switch (interfaceOrientation)
	{
		case UIInterfaceOrientationLandscapeLeft:
		{
			return AVCaptureVideoOrientationLandscapeLeft;
		}
			
		case UIInterfaceOrientationLandscapeRight:
		{
			return AVCaptureVideoOrientationLandscapeRight;
		}
			
      default:
		case UIInterfaceOrientationPortrait:
		{
			return AVCaptureVideoOrientationPortrait;
		}
			
		case UIInterfaceOrientationPortraitUpsideDown:
		{
			return AVCaptureVideoOrientationPortraitUpsideDown;
		}
	}
}

CGPathRef DTAVMetadataMachineReadableCodeObjectCreatePathForCorners(
                               AVCaptureVideoPreviewLayer *previewLayer,
                     AVMetadataMachineReadableCodeObject *barcodeObject)
{
   AVMetadataMachineReadableCodeObject *transformedObject =
   (AVMetadataMachineReadableCodeObject *)
   [previewLayer
    transformedMetadataObjectForMetadataObject:barcodeObject];
	
	// new mutable path
   CGMutablePathRef path = CGPathCreateMutable();
	
	// first point
   CGPoint point;
   CGPointMakeWithDictionaryRepresentation((__bridge CFDictionaryRef)
                                           transformedObject.corners[0],
                                           &point);
   CGPathMoveToPoint(path, NULL, point.x, point.y);
	
	// second point
   CGPointMakeWithDictionaryRepresentation((__bridge CFDictionaryRef)
                                           transformedObject.corners[1],
                                           &point);
   CGPathAddLineToPoint(path, NULL, point.x, point.y);
	
	// third point
   CGPointMakeWithDictionaryRepresentation((__bridge CFDictionaryRef)
                                           transformedObject.corners[2],
                                           &point);
   CGPathAddLineToPoint(path, NULL, point.x, point.y);
	
	// fourth point
   CGPointMakeWithDictionaryRepresentation((__bridge CFDictionaryRef)
                                           transformedObject.corners[3],
                                           &point);
   CGPathAddLineToPoint(path, NULL, point.x, point.y);
	
	// and back to first point
   CGPathCloseSubpath(path);
	
   return path;
}

File: Passbook/MovieNight/Swift/DTAVFoundationFunctions.swift
//
//  DTAVFoundationFunctions.swift
//  MovieNight
//
//  Created by Geoff Breemer on 12/10/14.
//  Copyright (c) 2014 Cocoanetics. All rights reserved.
//

import AVFoundation
import UIKit

// helper function to convert interface orienation to correct video capture orientation
func DTAVCaptureVideoOrientationForUIInterfaceOrientation(interfaceOrientation: UIInterfaceOrientation) -> AVCaptureVideoOrientation
{
    switch (interfaceOrientation)
        {
    case .LandscapeLeft:
        return .LandscapeLeft;
    case .LandscapeRight:
        return .LandscapeRight;
    case .Portrait:
        return .Portrait;
    case .PortraitUpsideDown:
        return .PortraitUpsideDown;
    case .Unknown:
        return .Portrait
    }
}

func DTAVMetadataMachineReadableCodeObjectCreatePathForCorners(
    previewLayer: AVCaptureVideoPreviewLayer,
    barcodeObject: AVMetadataMachineReadableCodeObject) -> CGPathRef
{
    let transformedObject: AVMetadataMachineReadableCodeObject? = previewLayer.transformedMetadataObjectForMetadataObject(barcodeObject) as? AVMetadataMachineReadableCodeObject
    
    // new mutable path
    var path: CGMutablePathRef = CGPathCreateMutable()
    
    // first point
    var point: CGPoint = CGPointZero
    CGPointMakeWithDictionaryRepresentation(transformedObject!.corners[0] as NSDictionary, &point)
    CGPathMoveToPoint(path, nil, point.x, point.y);
    
    // second point
    CGPointMakeWithDictionaryRepresentation(transformedObject!.corners[1] as NSDictionary, &point)
    CGPathAddLineToPoint(path, nil, point.x, point.y)
    
    // third point
    CGPointMakeWithDictionaryRepresentation(transformedObject!.corners[2] as NSDictionary, &point)
    CGPathAddLineToPoint(path, nil, point.x, point.y)
    
    // fourth point
    CGPointMakeWithDictionaryRepresentation(transformedObject!.corners[3] as NSDictionary, &point)
    CGPathAddLineToPoint(path, nil, point.x, point.y)
    
    // and back to first point
    CGPathCloseSubpath(path)
    
    return path
}